lib/Transforms/YosysOptimizer/YosysOptimizer.cpp

```diff
@@ -229,6 +229,44 @@ LogicalResult convertOpResults(secret::GenericOp op,
   return success();
 }
 
+void constantPassThrough(secret::GenericOp genericOp,
+                         mlir::IRRewriter &builder) {
+  builder.setInsertionPointToStart(genericOp.getBody());
+  genericOp.getBody()->walk([&](Operation *op) -> WalkResult {
+    for (Value operand : op->getOperands()) {
+      if (operand.getDefiningOp() &&
+          isa<arith::ConstantOp>(operand.getDefiningOp())) {
+        Region *operandRegion = operand.getParentRegion();
+        if (operandRegion && !genericOp.getRegion().isAncestor(operandRegion)) {
```

Comment:
Oops - this is also WIP - I did not actually handle this case yet. This branch of the code never does anything.

Reply:
(I had half implemented this case to walk all the operands, but since this pattern runs after capturing ambient scope, I was only interested in writing the block argument case. I'll fix this case too if I move this to run generically in secret patterns)

Reply:
my bad, reviewing too early :)